crates/resource/src/shape_manager.rs

```diff
@@ -198,10 +198,18 @@ impl ShapeManager {
 
         let slp_table = try!(drs_file.find_table(DrsFileType::Slp)
             .ok_or(ErrorKind::NoSlpTableInDrs(shape_key.drs_key)));
-        let slp_contents = try!(slp_table.find_file_contents(*shape_key.slp_id)
-            .ok_or(ErrorKind::SlpNotFound(shape_key.drs_key, *shape_key.slp_id)));
-
-        let slp = try!(SlpFile::read_from(&mut io::Cursor::new(slp_contents), *shape_key.player_color));
+        
+        let mut slp = SlpFile::new(0u8);
```

Comment:
Instead of making this a `mut` local would you please do this:

```rust
let slp = match slp_table.find_file_contents(*shape_key.slp_id) { ... };
Shape::load_from(&slp, ...)
```